### PR TITLE
MM-45161: Fix incorrect version check

### PR DESCRIPTION
**Pull request by agnivade**
The server version isn't a perfectly formed semver string.
We prune the other parts of the string before parsing it.

Fixes https://github.com/mattermost/mmctl/issues/528

https://mattermost.atlassian.net/browse/MM-45161

### CODE REVIEW

File: commands/init_test.go [hunk 1]
```diff
@@ -67,6 +67,13 @@ func TestCheckVersionMatch(t *testing.T) {
 			Expected:      false,
 			ErrExpected:   true,
 		},
+		{
+			Name:          "bad version",
+			Version:       "1.2.3",
+			ServerVersion: "1.2",
```
Comment (isacikgoz) [inline commands/init_test.go:73]: nit: technically 1.2 should be a valid version IMO. Not a big deal though as we probbably never return a version like this. Right?

Comment (agnivade) [reply commands/init_test.go:73]: You are right .. but the code parses 3 parts separated by ".". I think for our purposes this is good to keep. More like a test to lock in the behavior from server.

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks!

**Comment by agnivade** [comment]
@amyblais - This and https://github.com/mattermost/mmctl/pull/526 needs to be cherry-picked to 7.0 and a dot release needs to be done.

**Comment by amyblais** [comment]
Needs a manual cherry-pick.

**Comment by agnivade** [comment]
That's a weird one. @isacikgoz - cherry-picks work in the mmctl repo right?

**Pull request merged**